File: recursion/python/factorial.py
def factorial_iterative(n):
    val = 1
    for i in range(1, n + 1):
        val *= i

    return val


def factorial_recursive(n):
    if n == 0:
        # recursive case
        return 1
    else:
        # factorial case
        return factorial_recursive(n - 1) * n


if __name__ == '__main__':
    print(factorial_iterative(5))
    print(factorial_recursive(5))

File: recursion/javascript/factorial.js
// factorial function

// iterative solution
function factorialIterative(num) {
    let total = 1;

    for (let i = 1; i <= num; i++) {
        total *= i;
    }

    return total;
}

// recursive solution
function factorialRecursive(num) {
    if (num === 0) {
        // base case
        return 1;
    } else {
        // recursive case
        return num * factorialRecursive(num - 1);
    }
}

console.log(factorialIterative(5));
console.log(factorialRecursive(5));
